lumen/ai/ui.py

```diff
@@ -577,6 +577,11 @@ def _add_outputs(self, exploration: Column, outputs: list[LumenOutput], memory:
             else:
                 exploration.insert(0, sql_pane)
 
+        if isinstance(outputs[-1], str) and outputs[-1] == "__error__":
+            outputs.pop()
+            self._explorations.pop(-1)
```

Comment:
Did you test this? It breaks things in like 10 different ways.

Reply:
1. If we already generated some outputs and then errored it removes those outputs
2. If we were adding to an existing exploration this removes the old exploration
3. This pops the last exploration but we may be adding onto an existing one.
4. This only pops the explorations, there's also the conversations, titles, and contexts that need to be modified, which means everything will be out of sync after this.